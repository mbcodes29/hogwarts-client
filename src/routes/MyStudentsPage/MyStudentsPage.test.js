import React from 'react';
import ReactDOM from 'react-dom';
import MyStudentsPage from './MyStudentsPage'
import {BrowserRouter} from 'react-router-dom'
import renderer from "react-test-renderer"

it('renders without crashing',() => {
    const div = document.createElement('div')
    ReactDOM.render(
    <BrowserRouter>
        <MyStudentsPage />
    </BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
})

it('renders the UI as expected', () => {
    const tree = renderer
      .create(
      <BrowserRouter>
        <MyStudentsPage />
      </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });