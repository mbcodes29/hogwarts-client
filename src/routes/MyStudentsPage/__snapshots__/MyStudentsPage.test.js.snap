// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div>
  <nav
    className="Header"
  >
    <h1
      className="headerlink"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Create A Hogwarts Student!
      </a>
    </h1>
    <div
      className="Header__not-logged-in"
    >
      <a
        className="registerlink"
        href="/register"
        onClick={[Function]}
      >
         Create an account 
      </a>
      <p />
      <a
        className="loginlink"
        href="/login"
        onClick={[Function]}
      >
         Log in 
      </a>
    </div>
  </nav>
  <nav>
    <ul
      className="navbar"
    >
      <li
        className="nav-links"
      >
        <a
          className="navbarlink"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="nav-links"
      >
        <a
          className="navbarlink"
          href="/create-student"
          onClick={[Function]}
        >
          Create
        </a>
      </li>
      <li
        className="nav-links"
      >
        <a
          className="navbarlink"
          href="/my-students"
          onClick={[Function]}
        >
          My Students
        </a>
      </li>
    </ul>
  </nav>
  <section
    className="my-students"
  >
    <h1
      className="my-students-h1"
    >
      My Students
    </h1>
    <section
      id="students"
    />
  </section>
</div>
`;
